### PR TITLE
Add stats to BetaRowsetReader

### PR DESCRIPTION
**Pull request by chaoyli**
When doing schema change/compaction, only statistics of this RowsetReader is necessary.

### CODE REVIEW

File: be/src/olap/rowset/beta_rowset_reader.h [hunk 1]
```diff
@@ -53,6 +53,9 @@ class BetaRowsetReader : public RowsetReader {
     BetaRowsetSharedPtr _rowset;
 
     RowsetReaderContext* _context;
+    OlapReaderStatistics _owned_stats;
+    OlapReaderStatistics* _stats = &_owned_stats;
```
Comment (imay) [inline be/src/olap/rowset/beta_rowset_reader.h]: Do we really need this pointer? I think a local variable is enough

**Review by imay — CHANGES_REQUESTED** [review]
You should relate this PR to an ISSUE

**Comment by chaoyli** [comment]
#2072

**Comment by kangpinghuang** [comment]
You should add a description for this pr

File: be/src/olap/rowset/beta_rowset_reader.cpp [hunk 2]
```diff
@@ -38,7 +38,14 @@ OLAPStatus BetaRowsetReader::init(RowsetReaderContext* read_context) {
...
     StorageReadOptions read_options;
-    read_options.stats = _context->stats;
+    if (_context->stats == nullptr) {
+        // schema change/compaction should use owned_stats
+        // When doing schema change/compaction,
+        // only statistics of this RowsetReader is necessary.
+        read_options.stats = &_owned_stats;
+    } else {
+        read_options.stats = _context->stats;
+    }
```
Comment (kangpinghuang) [inline be/src/olap/rowset/beta_rowset_reader.cpp]:
```suggestion
   if (_context->stats != nullptr) {
       read_options.stats = _context->stats;
   }
```

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**